pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -668,7 +668,9 @@ func (r *ReconcileAWSFederatedAccountAccess) cleanFederatedRoles(reqLogger logr.
 					return err
 				}
 			}
-			if *policy.PolicyName == federatedRoleCR.Spec.AWSCustomPolicy.Name {
+
+			awsCustomPolicyname := federatedRoleCR.Spec.AWSCustomPolicy.Name + "-" + uidLabel
+			if *policy.PolicyName == awsCustomPolicyname {
```

Comment:
I did a quick check in this file alone for other instances of `federatedRoleCR.Spec.AWSCustomPolicy.Name` and found one in `deleteNonAttachedCustomPolicy` - can we ensure that the naming concatenation doesn't need to be done there as well? 

Would potentially be worth breaking `federatedRoleCR.Spec.AWSCustomPolicy.Name + "-" + uidLabel` into its own function?